### PR TITLE
GEODE-10151: Making geode-for-redis optional for a geode server

### PR DESCRIPTION
**Pull request by upthewaterspout**
Removing the geode-for-redis project and it's dependencies from the default
classpath of a geode server. Before this change, geode-for-redis was installed
in the lib directory of the server along with it's dependencies, and it was not
easy to run without it.

Creating a distribution for geode-for-redis that can be added to the classpath
of a server if the user chooses. This distribution is currently being built
in the geode-for-redis/build/distributions directory, as well as placed
as a zip file in the tools/Modules/ directory of the overall geode
distribution. This distribution only contains dependencies that are not
already on the classpath of a geode server.

To generate the redis distribution run
./gradlew geode-for-redis:installDist

or for the whole geode distribution, including the redis zip file run
./gradlew geode-assembly:installDist

See the gradle documentation on the distribution plugin for more details.

The redis tests that previously required a GEODE_HOME to run now also require
a GEODE_FOR_REDIS_HOME directory that points into an installed distribution of
geode-for-redis. The tests add the geode-for-redis jars into the classpath using this
environment variable. When running tests through gradle we set the
GEODE_FOR_REDIS_HOME environment variable automatically.

### CODE REVIEW

File: geode-assembly/build.gradle [hunk 1]
```diff
@@ -524,6 +523,7 @@ distributions {
         from { project(':extensions:geode-modules-assembly').distTcServer30 }
         from { project(':extensions:geode-modules-assembly').distTomcat }
         from { project(':extensions:geode-modules-assembly').distAppServer }
+        from { project(':geode-for-redis').distZip }
```
Comment (rhoughton-pivot) [inline geode-assembly/build.gradle:526]: Being a *nix user, I prefer the archives be in tgz format. Thoughts?

**Review by onichols-pivotal — APPROVED** [review]

File: geode-assembly/build.gradle (hunk 1 above)
Comment (upthewaterspout) [reply geode-assembly/build.gradle:526]: Well, those other jobs just above this one like `distTomcat`, `distAppServer` etc. produce zip files. So I was trying to match the archive format of all of the other stuff in this modules directory. Make sense?

**Comment by rhoughton-pivot** [comment]
Should we name the archive `tools/Modules/Apache_Geode_Modules-1.15.0-redis.zip` for alignment with the other modules in that directory?

**Comment by jdeppe-pivotal** [comment]
I noticed a `geo-0.8.0.jar` jar in the `geode-for-redis` zip. I'm not sure why we have this dependency, but I think it can be removed. Servers start without it, integration tests run fine and `./gradlew geode-for-redis:dependencies` doesn't show it.

**Comment by jdeppe-pivotal** [comment]
`netty-handler` is pulling in `netty-tcnative` which we don't need and could be excluded.

**Review by jdeppe-pivotal — APPROVED** [review]
Looks good from what I can tell. Did some basic tests starting up with `gfsh` and using a classpath to reference the jars.

**Comment by upthewaterspout** [comment]
> I noticed a `geo-0.8.0.jar` jar in the `geode-for-redis` zip. I'm not sure why we have this dependency, but I think it can be removed. Servers start without it, integration tests run fine and `./gradlew geode-for-redis:dependencies` doesn't show it.
Interesting. I didn't add these, but it does look like maybe they aren't necessary? I pushed a change to remove them.

**Comment by upthewaterspout** [comment]
> Should we name the archive `tools/Modules/Apache_Geode_Modules-1.15.0-redis.zip` for alignment with the other modules in that directory?

We could. It seems like we're a little inconsistent with naming - for example I see geode-pulse-0.0.0.war. That being the case, I'd rather stick with the gradle default name if that works for you.

**Review by rhoughton-pivot — APPROVED** [review]

**Pull request merged**